plugins/chunter-resources/src/components/Channel.svelte

```diff
@@ -114,14 +116,105 @@
   }
 
   let newMessagesPos: number = -1
+
+  function isOtherDay (time1: Timestamp, time2: Timestamp) {
+    return getDay(time1) !== getDay(time2)
+  }
+  let messagesDays: Timestamp[] | undefined = []
+  $: {
+    let firstDate
+    if (messages?.[0]) {
+      firstDate = getDay(messages[0].createOn)
+    }
+
+    messagesDays = messages?.reduce(
+      (prev, cur) => {
+        if (isOtherDay(prev[prev.length - 1], cur.createOn)) {
+          prev.push(getDay(cur.createOn))
+        }
+        return prev
+      },
+      firstDate ? [firstDate] : []
+    )
+  }
+
+  function handleJumpToDate (e: CustomEvent<any>) {
+    const date = e.detail.date
+    if (!date) {
+      return
+    }
+
+    let closestDate: Timestamp | undefined = messagesDays?.reduceRight((prev, cur) => {
+      if (prev < date) return cur
+      if (cur < date) return prev
+      if (cur - date < prev - date) return cur
+      else return prev
+    })
+    if (closestDate && closestDate < date) closestDate = undefined
+
+    if (closestDate) {
+      scrollToDate(closestDate)
+    }
+  }
+
+  const pinnedHeight = 30
+  const headerHeight = 70
+  function scrollToDate (date: Timestamp) {
+    let offset = date && document.getElementById(date.toString())?.offsetTop
+    if (offset) {
+      offset = offset - headerHeight
+      if (pinnedIds.length > 0) offset = offset - pinnedHeight
+      div?.scrollTo({ left: 0, top: offset })
+    }
+  }
+
+  let up: boolean | undefined = true
+  let showFixed: boolean | undefined = true
+  let selectedDate: Timestamp = messagesDays[messagesDays.length - 1]
+  function handleScroll () {
+    up = div && div.scrollTop === 0
+
+    const upperVisible = getFirstVisible()
+    if (upperVisible) {
+      selectedDate = parseInt(upperVisible.id)
+      showFixed = false
+      if (!selectedDate) {
+        selectedDate = getDay(parseInt(upperVisible.getAttribute('data-date')))
+        showFixed = true
+      }
+    }
+  }
+
+  function getFirstVisible (): HTMLElement {
+    const clientRect = div.getBoundingClientRect()
+    for (let i = div?.childElementCount; i >= 0; i--) {
```

Comment:
Better way add some class to `JumpToDateSelector`, `dateSelector` for example, find `div.getElementsByClassName('dateSelector')`, and use it